nautobot/extras/constants.py

```diff
@@ -30,6 +30,14 @@
     "time_limit",
 )
 
+
+# Job field length limits
+JOB_MAX_NAME_LENGTH = 100  # TODO this should really be a more general-purpose constant, like NAME_MAX_LENGTH
+JOB_MAX_SLUG_LENGTH = 320  # 16 source, 100 GitRepository.name, 100 module_name, 100 job_class_name
+JOB_MAX_GROUPING_LENGTH = 255
+JOB_MAX_SOURCE_LENGTH = 16  # "git", "local", "plugins"
```

Comment:
`nit`

In your TODO you would call this `NAME_MAX_LENGTH` but you set them here to `JOB_MAX_NAME_LENGTH`.

I think a reasonable syntax would be:

`MODEL_ATTR_MAX/MIN_CONSTRAINT` so `JOB_NAME_MAX_LENGTH` because for example if we had a min length the sorting would look nice `JOB_NAME_MIN_LENGTH`.

Reply:
I'd prefer that as well but I'm following the existing pattern in this file - see JOB_LOG_MAX_GROUPING_LENGTH, etc. just below.